ecommerce/permissions.py

```diff
@@ -31,3 +32,18 @@ def has_permission(self, request, view):
                 request.data,
             )
             return False
+
+
+class IsLoggedInUser(BasePermission):
+    """
+    Confirms that the username in the request body is the same as the logged in user's.
+    """
+
+    def has_permission(self, request, view):
+        """
+        Returns true if the username in the request body matches the logged in user.
+        """
+        try:
+            return request.data['username'] == get_social_username(request.user)
```

Comment:
The only username we expose to the user is the edX one. For example `SETTINGS.user.username` is the edX username of the logged in user and it's also the one used in `/api/v0/profiles/<username>/`. This is to not confuse the two in our API